Migrate from CircleCI to Github Actions

enpaul opened the pull request:
## Description

<!-- If this is a deploy PR, please append `?template=deploy.md` to the current URL -->

Works towards #1162 

Summary of changes:

* Migrate `build_prod` CircleCI job to the `build-prod` Github Actions job
* Migrate `build_dev` CircleCI job into the Github Actions jobs `build-dev` and `test` to report test failures independently and improve digestibility of the test results.
* Add `lint-actions` job for linting the Github Actions config itself
* Add triggers for running CI on push to `develop` and `prod` branches
* Add triggers for running CI when a PR is opened or synchronized
* Remove CircleCI config
* Remove CircleCI job that uses `safety` to check local repository since that functionality is [already centralized in freedomofpress/fpf-www-projects](https://github.com/freedomofpress/fpf-www-projects/blob/main/tools/safety_check.py#L658).
* Remove CircleCI job that uses `npm audit` to check local repository since that functionality is [already centralized in freedomofpress/fpf-www-projects](https://github.com/freedomofpress/fpf-www-projects/blob/main/tools/npm-audit-data.json#L4).
* Update README to include status badges for new CI jobs on `prod` and `develop` branches

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Vulnerabilities update
- [x] Config changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires an admin update after deploy
- [ ] Includes a database migration removing  or renaming a field


## Testing

See CI on this PR for test runs

> [!NOTE]
> Since the CircleCI config is still active for this repository but has been removed from this branch, job failures reported by CircleCI are expected. Only Github Actions jobs are expected to succeed on this branch.

enpaul committed:
Remove redundant security check jobs

These jobs already run in the centralized security check repo so there's no need to repeat
their execution here

enpaul committed:
Add github actions jobs for building prod and dev environments

Add job for linting local actions workflows
Add initial scaffolding for building environments in github actions
Add job for building development images and publishing them as actions artifacts
Add job for automatically cleaning dev image artifacts
Add job for building prod images and running sanity checks on them

enpaul committed:
Add github actions job for running dev tests

Add matrix for running various targets on the fly using built images
Fix cleanup job to always run even if test job fails
Add job triggers for prod and develop branches

enpaul committed:
Remove circleci config

enpaul committed:
Update readme to display CI status badges

mig5 reviewed (approved):
It's looking good to me! Great to see the Actions passing!

I think it will be good to have one of the web team double-check the Actions Checks output on this PR to make sure all the build / 'make test' steps are returning the expected output (I'm less familiar with this repo's CI)

enpaul committed:
Misc fixes and cleanup

Remove duplicate permissions from lint-actions job
Update docker version to use workflow constant
Update curl version to use workflow constant

chigby reviewed (commented):
I think this is looking really great. The organization is, of course, different from what I'm more used to in CircleCI, but as I've said before, I really like what I'm seeing here in terms of de-duplicated code and visible test output. :)
